src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -670,11 +673,18 @@ private static DataContract CreateDataContract(int id, RuntimeTypeHandle typeHan
                         {
                             if (type == null)
                                 type = Type.GetTypeFromHandle(typeHandle);
-#if !NET_NATIVE
                             type = UnwrapNullableType(type);
-#endif
                             type = GetDataContractAdapterType(type);
-#if !NET_NATIVE
+
+#if NET_NATIVE
+                            dataContract = DataContract.GetDataContractFromGeneratedAssembly(type);
+                            if (dataContract != null)
+                            {
+                                AssignDataContractToId(dataContract, id);
+                                return dataContract;
+                            }
```

Comment:
Is it necessary to call AssignDataContractToId here? We call this method after calling CreateDataContract() already.